lib/internal/repl.js

```diff
@@ -64,27 +69,14 @@ function setupHistory(repl, historyPath, ready) {
   var writing = false;
   var pending = false;
   repl.pause();
-  fs.open(historyPath, 'a+', oninit);
+  fs.readFile(historyPath, {encoding: 'utf8', flag: 'a+'}, ondata);
```

Comment:
Per [the docs](https://iojs.org/api/fs.html#fs_fs_open_path_flags_mode_callback), I'm using "a+" for "create file if it doesn't exist, retain contents for reading if it does."